provisioner/clusterpy.go

```diff
@@ -442,28 +446,37 @@ func createOrUpdateClusterStack(ctx context.Context, config channel.Config, clus
 	return outputs, nil
 }
 
-func filterSubnets(allSubnets []*ec2.Subnet, subnetIds []string) ([]*ec2.Subnet, error) {
-	desiredSubnets := make(map[string]struct{})
-	for _, id := range subnetIds {
-		desiredSubnets[id] = struct{}{}
+func filterSubnets(subnets []*ec2.Subnet, filter func(*ec2.Subnet) bool) []*ec2.Subnet {
+	var filtered []*ec2.Subnet
+	for _, subnet := range subnets {
+		if filter(subnet) {
+			filtered = append(filtered, subnet)
+		}
 	}
 
-	var result []*ec2.Subnet
-	for _, subnet := range allSubnets {
-		subnet := *subnet
-		subnetID := aws.StringValue(subnet.SubnetId)
-		_, ok := desiredSubnets[subnetID]
-		if ok {
-			result = append(result, &subnet)
-			delete(desiredSubnets, subnetID)
+	return filtered
+}
+
+func subnetIDIncluded(ids []string) func(*ec2.Subnet) bool {
+	return func(subnet *ec2.Subnet) bool {
+		for _, id := range ids {
+			if aws.StringValue(subnet.SubnetId) == id {
+				return true
+			}
 		}
+
+		return false
 	}
+}
 
-	if len(desiredSubnets) > 0 {
-		return nil, fmt.Errorf("invalid or unknown subnets: %s", desiredSubnets)
+func notCustomSubnet(subnet *ec2.Subnet) bool {
```

Comment:
I would suggest to use the normal version of a predicate `isCustomSubnet` for the function and negate it on the caller side. This just reads a bit weird.